extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/utils/ValidatorUtils.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2024 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.ext.wrapper.utils;
+
+import jakarta.validation.ConstraintViolationException;
+import jakarta.validation.Validation;
+import jakarta.validation.ValidatorFactory;
+import lombok.val;
+
+public class ValidatorUtils {
+    private static final ValidatorFactory FACTORY = Validation.buildDefaultValidatorFactory();
+
+    public static void validate(Object object) {
+        val validator = FACTORY.getValidator();
+        val constraintViolations = validator.validate(object);
+        if (!constraintViolations.isEmpty()) {
+            throw new ConstraintViolationException("Failed to validate", constraintViolations);
```

Comment:
Don't you want to throw a derivate of WebApplicationException here, so that the message reaches the User?

Reply:
They're all bad on the client side.

throw BadRequest:

```
de.sovity.edc.client.gen.ApiException: Message: Bad Request
HTTP response code: 400
HTTP response body: {
"servlet":"EDC-management",
"message":"Bad Request",
"url":"/api/management/wrapper/ui/pages/content-agreement-page/Y2QtYWx3YXlzLXRydWUtYXNzZXQtMA%3D%3D%3AYXNzZXQtMA%3D%3D%3AYTJlZDM1YTUtNWM1NC00MzI1LWJkNTAtMWNlOGVhOTMxNjIw/terminate",
"status":"400"
}
```

throw ConstraintViolationException:

```
de.sovity.edc.client.gen.ApiException: Message: Server Error
HTTP response code: 500
HTTP response body: {
"servlet":"EDC-management",
"message":"Internal Server Error",
"url":"/api/management/wrapper/ui/pages/content-agreement-page/Y2QtYWx3YXlzLXRydWUtYXNzZXQtMA%3D%3D%3AYXNzZXQtMA%3D%3D%3AZDBhNDkwOTAtMWM3Yi00ZDNmLWJmODgtYmM1MzAyMWQ3MGNk/terminate",
"status":"500"
}
HTTP response headers: {cache-control=[must-revalidate,no-cache,no-store], content-length=[263], content-type=[application/json], server=[Jetty(11.0.15)]}
```

throw WebApplicationException:
```
de.sovity.edc.client.gen.ApiException: Message: Server Error
HTTP response code: 500
HTTP response body: {
"servlet":"EDC-management",
"message":"Internal Server Error",
"url":"/api/management/wrapper/ui/pages/content-agreement-page/Y2QtYWx3YXlzLXRydWUtYXNzZXQtMA%3D%3D%3AYXNzZXQtMA%3D%3D%3AZjg0ZjdjMjYtZjEzNy00NDFiLWJlOWQtODIxYjIzOGQ2MTZj/terminate",
"status":"500"
}
HTTP response headers: {cache-control=[must-revalidate,no-cache,no-store], content-length=[263], content-type=[application/json], server=[Jetty(11.0.15)]}
```

What I'd like to do is something like:
```
@Override
    public Response terminateContractAgreement(
        String contractAgreementId,
        ContractTerminationRequest contractTerminationRequest
    ) {

        try {
            validate(contractTerminationRequest);
        } catch (ValidationException e) {
            return Response.status(Response.Status.BAD_REQUEST).entity(e).build();
        }

        return Response.ok(
            dslContextFactory.transactionResult(dsl ->
                contractAgreementTerminationApiService.terminate(dsl, contractAgreementId, contractTerminationRequest))
        ).build();

    }
```

But this causes issues in the generator, where it now can't see the type that's returned: it think it returns nothing...